org.eclipse.jdt.core/dom/org/eclipse/jdt/internal/core/dom/rewrite/ASTRewriteAnalyzer.java

```diff
@@ -665,8 +666,13 @@ private int rewriteList(
 						if (separatorState == EXISTING) {
 							updateIndent(prevMark, currPos, i, editGroup);
 						}
-
-						doTextInsert(currPos, node, getNodeIndent(i), true, editGroup); // insert node
+							doTextInsert(
```

Comment:
this function is a much used function and we should avoid node-specific checks here. need to hoist this part of checking for StringTeemplate at the caller side isolated in that node - or find another way to do this rewrite.

Reply:
Done.